botocore/client.py

```diff
@@ -606,8 +608,21 @@ def _create_endpoint(
             resolved, region_name, endpoint_url
         )
         if endpoint_url is None:
-            # Use the sslCommonName over the hostname for Python 2.6 compat.
-            hostname = resolved.get('sslCommonName', resolved.get('hostname'))
+            sslCommonName = resolved.get('sslCommonName')
+            hostname = resolved.get('hostname')
+            is_disabled = ensure_boolean(
+                os.environ.get('BOTO_DISABLE_COMMONNAME', False)
+            )
+            if not is_disabled and sslCommonName is not None:
+                warnings.warn(
+                    f'The {service_name} client is currently using a '
+                    f'deprecated endpoint: {sslCommonName}. In the next '
+                    f'minor version this will be moved to {hostname}. '
+                    'See https://github.com/boto/botocore/issues/2705 '
+                    'for more details.',
+                    category=FutureWarning,
+                )
+                hostname = sslCommonName
```

Comment:
Did we confirm this warning does not trigger for the CLI?

Reply:
so it should never be triggered even if the environment variable isn't set? that PR doesn't suppress any warnings so I guess I'm wondering how is this deprecation supposed to work at all in the CLI. It will still hit this code path, but will silently change it back to the common name with the mapping. let me double check again

Reply:
looks like we're good since `endpoint_url` will never be `None` in the CLI v1 since we set it in on `parsed_globals`

Reply:
feel free to test yourself as well, but i didn't get any warnings